lib/modules/hosts/wikipedia.js

```diff
@@ -6,10 +6,40 @@ export default {
 	name: 'wikipedia',
 	domains: ['wikipedia.org', 'wikipedia.com'],
 	logo: 'https://en.wikipedia.org/static/favicon/wikipedia.ico',
-	detect: ({ href }) => (/^https?:\/\/([A-z]{2}|simple)?\.(?:m\.)?wikipedia\.(?:com|org)\/wiki\/([^#\?]+)/i).exec(href),
-	async handleLink(href, [, country, article]) {
+	detect: url => {
+		return {
+			article: url.pathname.substr(6), // remove "/wiki/"
+			language: url.host.split('.')[0],
+			hash: url.hash.substr(1),
+		};
+	},
+	async handleLink(href, { language, article, hash }) {
+		// default to section 0
+		let sectionId = 0;
+
+		// If we have a hash, try and look up its section number
+		if (hash !== '') {
+			const sections = await ajax({
+				url: `https://${language}.wikipedia.org/w/api.php?action=parse&format=json&prop=sections&page=${article}&origin=*`,
+				type: 'json',
+				headers: {
+					// See https://www.mediawiki.org/wiki/API:Main_page#Identifying_your_client
+					'Api-User-Agent': 'Reddit-Enhancement-Suite/5.0 ( https://github.com/honestbleeps/Reddit-Enhancement-Suite/ )',
```

Comment:
How about
import * as Metadata from '../core/metadata';

`Reddit-Enhancement-Suite/${Metadata.version} ( ${Metadata.updatedURL} )`

or even better, add to metadata.js: `export const homepageURL = packageInfo.homepage;` and for the user-agent here, `Reddit-Enhancement-Suite/${Metadata.version} ( ${Metadata.homepageURL} )`